File: nebo-fontends/apps/storefont/app/types/category.ts
import React from "react";

export type GroupCategory = {
  id: number;
  icon: React.ReactNode;
  name: string;
  description: string;
};

export type Category = {
  id: number;
  name: string;
  group_id: number;
  created_on: string;
  updated_on: string;
};

export type CategoryByGroup = {
  group_id: number;
  categories: Category[];
};

export type CategoryByGroupFilterRequest = {
  owner?: boolean;
};

File: nebo-services/service/template/src/main/java/com/nebo/template/infrastructures/domain/model/Category.java
package com.nebo.template.infrastructures.domain.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;

@Setter
@Getter
@Table(name = "categories")
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private long userId;
    private int groupId;
    private String name;
    private Instant createdOn;
    private Instant updatedOn;
}
